### DISCUSSION TITLE
Matic Plasma client created, but does not initialise on mainnet

### DISCUSSION THREAD

**Discussion by naman-modi** [0xPolygon/matic.js#313] [Q&A]
```
    const networkInstance = new Network(mainnet, v1);
    const matic = new MaticPlasmaClient({
      network: mainnet,
      version: v1,
      parentProvider: new HDWalletProvider(privateKey, config.parent.rpc),
      maticProvider: new HDWalletProvider(privateKey, config.child.rpc),
      parentDefaultOptions: { from: userAddress },
      maticDefaultOptions: { from: userAddress },
    });
    
    await matic.initialize();
```

In the above code:
-  `config.parent.rpc=https://goerli.infura.io/v3/........`
- `config.child.rpc=https://polygon-mumbai.g.alchemy.com/v2/.......` 
- `userAddress` and `privateKey` are correct.

The exact **error**:
`Error: Returned values aren't valid, did it run Out of Gas? You might also see this error if you are not using the correct ABI for the contract you are retrieving data from, requesting data from a block number that does not exist, or querying a node which is not fully synced.`

The `matic` object gets created, but `initialize()` does not work. Any thoughts to make it work?
(Note: the userAddress wallet has enough for Gas)

**Comment by ujjwalgupta94** [comment]
@naman-modi please use v3 of maticjs, the all issues has been fixed in v3. Here is doc link - https://maticnetwork.github.io/matic.js/docs/plasma/
